pkg/resources/validation/cluster/provisioningcluster.go

```diff
@@ -135,6 +147,36 @@ func validateCreatorID(request *webhook.Request, oldCluster, cluster *v1.Cluster
 	return nil
 }
 
+func (p *provisioningClusterValidator) validateClusterName(request *webhook.Request, response *webhook.Response, cluster *v1.Cluster) error {
+	if request.Operation != admissionv1.Create {
+		return nil
+	}
+
+	_, err := p.mgmtClusterClient.Get(cluster.Name, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
```

Comment:
Wouldn't it be an issue if the cluster name is already in use? Or are you just going to let kubernetes return that error?

Reply:
Rancher will create a provisioning cluster for every management cluster. That is, if a user creates an RKE1 cluster, for example, then the Rancher controller will create a provisioning cluster of the same name. We want to allow this.

Reply:
Oh I see, bc this validation runs on client requests to we have to allow it or the controller would not be able to do that.

Reply:
Then what is this get checking for exactly?

Reply:
If a management cluster exists with the same name as the provisioning cluster that is being created, then that is allowed because Rancher is doing that.

If no such management cluster exists, then the don't allow the provisioning cluster to be created because that means the user is trying to create a provisioning cluster with the "name format" of a management cluster. We don't want that to happen because there is Rancher logic that expects provisioning clusters with that name format to have come from an existing management cluster.

Reply:
@thedadams Based on these comments can you include this as code comments, I suspect this will come up again in the future.

Reply:
I augmented my comment below. Please let me know if you think more comments are necessary.

Reply:
Can you comment above the get call as well?

Reply:
Done.